### PR TITLE
1 dip

### PR DESCRIPTION
**Pull request by MargaretDuff**
## Describe your contribution
- I reordered and added explanations to the notebooks
- Added the correct license documents
- Fixed the scaling on the problem to match the explanation
- Fixed the singular value issue by changing the ranges in calculating the matrix A
I still haven't fixed the issue in reconstructing the values near the right-hand boundary at 1. Would appreciate someone else looking at it.

## Checklist when you are ready to request a review

- [X ] I have performed a self-review of my code
- [X ] I have created a new folder, containing my contributions which are in the form of jupyter notebooks, with any necessary supporting python files, and a LICENSE file.
- [X ] I have added a description of my contribution(s) to the top of my file(s)
- [ X] If publicly available, I have added a link to the dataset used near the top of my file(s)
- [ X] I have added the CIL version I ran with near the top of my file(s)
- [ X] The content of this Pull Request (the Contribution) is intentionally submitted for inclusion in CIL-User-Showcase.
- [X ] I confirm that the contribution does not violate any intellectual property rights of third parties.
- [X ] I confirm that I have added license headers to all of the files I am contributing (with a license of my choice)
- [X ] Change pull request label to 'Waiting for review' 

Note: for an example of a contribution, where a license header, description, data link and CIL version has been added, please
see: [example_contribution](https://github.com/TomographicImaging/CIL-User-Showcase/blob/main/example_contribution)

### CODE REVIEW

File: 1DIP/deriv2_cgls.ipynb [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
+    "#   Authored by:  Bill Lionheart (University of Manchester), Margaret DUff (STFC - UKRI)"
+   ]
+  },
   {
    "cell_type": "markdown",
    "id": "6f8acbea",
    "metadata": {},
    "source": [
-    "###1D inverse problem demo using deriv2 from regtools"
+    "# 1D inverse problem demo using deriv2 from regtools"
```
Comment (jakobsj) [inline 1DIP/deriv2_cgls.ipynb]: In the following line (cant seem to comment on it)  need to change SDV to SVD. Even better write "singular value decomposition (SVD)"

File: 1DIP/deriv2_cgls.ipynb [hunk 2]
```diff
@@ -186,229 +264,249 @@
...
+      " -1.23512311e-15 -4.30211422e-16  8.88178420e-16  9.15933995e-16\n",
+      " -4.37150316e-16 -1.20736754e-15 -5.68989300e-16  8.04911693e-16\n",
+      "  9.92261828e-16 -2.08166817e-16 -1.11022302e-15 -6.10622664e-16\n",
+      "  7.49400542e-16  1.08246745e-15 -1.38777878e-17 -1.01654796e-15\n",
+      " -7.59808882e-16  6.14092110e-16  1.10328413e-15  2.18575158e-16\n",
+      " -9.87057658e-16 -8.69096461e-16  4.18068358e-16  1.17354043e-15]\n"
+     ]
+    },
+    {
+     "ename": "TypeError",
```
Comment (jakobsj) [inline 1DIP/deriv2_cgls.ipynb]: There is an error showing up in the output here - needs to be resolved.

File: 1DIP/deriv2_cgls.ipynb [hunk 3]
```diff
@@ -59,27 +63,118 @@
...
+    "    K(s,t) = \\begin{cases} s(t-1)  ,  s <  t \\\\\n",
+    "              t(s-1)  ,  s \\geq t \\end{cases} $$\n",
+    "\n",
+    "and $$ \\int_0^1K(s,t)x(t)dt=b(s). $$\n",
+    "\n",
+    " For this notebook, consider the case\n",
+    "$$ b(s) = (s^3 - s)/6          , \\ \\  x(t) = t$$\n",
+    "\n",
+    " where the integral is disretised using the Galerkin method with orthonormal box functions, with interval size $1/n$.\n",
+    "\n"
```
Comment (jakobsj) [inline 1DIP/deriv2_cgls.ipynb]: Mention somewhere early on that the notebook will first set up and solve basic formulation of the problem using just python/numpy, and after that using CIL.

File: 1DIP/deriv2_cgls.ipynb [hunk 4]
```diff
@@ -418,27 +516,42 @@
...
-   "execution_count": 122,
+   "execution_count": null,
    "id": "15ff9d79",
    "metadata": {},
    "outputs": [],
    "source": [
     "ig = Aop.domain_geometry()\n",
     "L = IdentityOperator(ig)\n",
-    "# L=np.identity(n)\n",
     "alpha = 0.01\n",
```
Comment (jakobsj) [inline 1DIP/deriv2_cgls.ipynb]: This is a different value of the regularization parameter compared to what was used for the non-CIL tikhonov problem earlier in the notebook. Would be better to define alpha back then and reuse here to hopefully get the same solution without and with CIL and comment that CIL reproduces what was done without CIL.

**Review by jakobsj — CHANGES_REQUESTED** [review]
Many thanks @MargaretDuff for working through and polishing this notebook - it is looking very nice now and almost ready in my view. It seems an error occurred during the final run so not all output is generated, please can you look into this. I also made a couple of minor requests inline.

File: 1DIP/deriv2_cgls.ipynb [hunk 5]
```diff
@@ -1,11 +1,36 @@
...
+    "#\n",
+    "#      http://www.apache.org/licenses/LICENSE-2.0\n",
+    "#\n",
+    "#  Unless required by applicable law or agreed to in writing, software\n",
+    "#  distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+    "#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+    "#  See the License for the specific language governing permissions and\n",
+    "#  limitations under the License.\n",
+    "#\n",
+    "#   Authored by:  Bill Lionheart (University of Manchester), Margaret DUff (STFC - UKRI)"
```
Comment (jakobsj) [inline 1DIP/deriv2_cgls.ipynb]: I suggest we add a new line "Reviewed by: " or "Reviewed/edited by:" or similar and add names of those of us who reviewed/edited to keep the user contributor as the author.

Comment (MargaretDuff) [reply 1DIP/deriv2_cgls.ipynb]: Sounds good - will update this

File: 1DIP/deriv2_cgls.ipynb (hunk 3 above)
Comment (MargaretDuff) [reply 1DIP/deriv2_cgls.ipynb]: Done - thanks Jakob

File: 1DIP/deriv2_cgls.ipynb (hunk 1 above)
Comment (MargaretDuff) [reply 1DIP/deriv2_cgls.ipynb]: Done - thanks

File: 1DIP/deriv2_cgls.ipynb (hunk 2 above)
Comment (MargaretDuff) [reply 1DIP/deriv2_cgls.ipynb]: Thanks - done!

File: 1DIP/deriv2_cgls.ipynb (hunk 4 above)
Comment (MargaretDuff) [reply 1DIP/deriv2_cgls.ipynb]: Thanks Jakob - have changed this and added a comparison plot between the CIL and numpy solutions at the end of the notebook.

**Review by jakobsj — APPROVED** [review]
Looks good and great to see the solutions compared explicitly (and that they match!). Again would say to move "edited by" to new line, otherwise I suggest can be merged into @billlion's branch

**Comment by MargaretDuff** [comment]
Thanks @jakobsj, fixed! @billlion - could you merge them?

**Comment by billlion** [comment]
Thanks @MargaretDuff . I will have a look at that end effect!

**Comment by MargaretDuff** [comment]
Thank-you!

**Pull request merged**